File: CSharp/SubscriptionApi/SubscriptionApi/Subscriptions/Subscriptions.cs
using System.Collections.Generic;
using System.Linq;
using SubscriptionApi.ResponseBuilders;

namespace SubscriptionApi.Subscriptions
{
  public interface SubscriptionsModifyOperations
  {
    void AddNew(Subscription subscription);
    void Remove(string subscriptionId, SubscriptionStopEvents events);
  }

  public class Subscriptions : SubscriptionsModifyOperations
  {
    private readonly List<Subscription> _subscriptions = new List<Subscription>();
 
    public void AddNew(Subscription subscription)
    {
      subscription.ScheduleExpiry();
      _subscriptions.Add(subscription);
    }

    public void Remove(string subscriptionId, SubscriptionStopEvents events)
    {
      var subscription = _subscriptions.First(s => s.Has(subscriptionId));
      if (subscription != null)
      {
        subscription.ForceExpiry();
        _subscriptions.Remove(subscription);
      }
      else
      {
        events.NoSubscriptionToStopWithId(subscriptionId);
      }
    }
  }
}

File: Java/factories/_01_encapsulation_of_global_context/src/main/java/subscriptions/Subscriptions.java
package subscriptions;

import responseBuilders.SubscriptionStopEvents;

import java.util.ArrayList;
import java.util.List;

public class Subscriptions implements SubscriptionsModifyOperations {
    private final List<Subscription> _subscriptions = new ArrayList<>();

    @Override
    public void addNew(Subscription subscription) {
        subscription.scheduleExpiry();
        _subscriptions.add(subscription);
    }

    @Override
    public void remove(String subscriptionId, SubscriptionStopEvents events) {
        Subscription subscription = _subscriptions.stream().filter(s -> s.has(subscriptionId)).findFirst().get();
        if (subscription != null) {
            subscription.forceExpiry();
            _subscriptions.remove(subscription);
        } else {
            events.noSubscriptionToStopWithId(subscriptionId);
        }
    }
}
